net.c

```diff
@@ -521,25 +540,30 @@ int redisContextConnectUnix(redisContext *c, const char *path, const struct time
         return REDIS_ERR;
 
     c->connection_type = REDIS_CONN_UNIX;
-    if (c->unix_sock.path != path)
+    if (c->unix_sock.path != path) {
```

Comment:
Another one.... Looks like a pattern and this one leaks as well.

Reply:
Good catch on the leak, I'll update the logic!